packages/instant-meilisearch/src/adapter/search-request-adapter/filter-adapter.ts

```diff
@@ -1,93 +1,155 @@
 import type { Filter, SearchContext } from '../../types'
 
+const filterEscapeRegExp = /([\\"])/g
+
+function getValueWithEscapedBackslashesAndQuotes(value: string): string {
+  return value.replace(filterEscapeRegExp, '\\$1')
+}
+
 /**
- * Transform InstantSearch filter to Meilisearch compatible filter format.
+ * Transform InstantSearch [facet filter](https://www.algolia.com/doc/api-reference/api-parameters/facetFilters/)
+ * to Meilisearch compatible filter format.
  * Change sign from `:` to `=`
  * "facet:facetValue" becomes "facet=facetValue"
  *
  * Wrap both the facet and its facet value between quotes.
- * This avoid formating issues on facets containing multiple words.
+ * This avoids formatting issues on facets containing multiple words.
+ * Escape backslash \\ and quote " characters.
  *
  * 'My facet:My facet value' becomes '"My facet":"My facet value"'
  *
- * @param  {string} filter?
- * @returns {Filter}
+ * @param {string} filter
+ * @returns {string}
+ */
+function transformFacetFilter(filter: string): string {
+  const escapedFilter = getValueWithEscapedBackslashesAndQuotes(filter)
+  const colonIndex = escapedFilter.indexOf(':')
+  const attribute = escapedFilter.slice(0, colonIndex)
+  const value = escapedFilter.slice(colonIndex + 1)
+  return `"${attribute}"="${value}"`
+}
+
+// Matches first occurrence of an operator
+const numericSplitRegExp = /(?<!(?:[<!>]?=|<|>|:).*)([<!>]?=|<|>|:)/
+
+/**
+ * Transform InstantSearch [numeric filter](https://www.algolia.com/doc/api-reference/api-parameters/numericFilters/)
+ * to Meilisearch compatible filter format.
+ *
+ * 'price:5.99 TO 100' becomes '"price" 5.99 TO 100'
+ *
+ * 'price = 5.99' becomes '"price"=5.99'
+ *
+ * Wrap the attribute between quotes.
+ * Escape backslash (\\) and quote (") characters.
+ *
+ * @param {string} filter
+ * @returns {string}
  */
-function transformFilter(filter: string): string {
-  return filter.replace(/(.*):(.*)/i, '"$1"="$2"')
+function transformNumericFilter(filter: string): string {
+  // TODO: What if escape facet values is enabled?
```

Comment:
If we merge your PR, those questions will likely never be answered... What can we do about them?

Reply:
Keep the TODO and you can create a new issue :)